source/octf/utils/table/Table.cpp

```diff
@@ -141,87 +331,102 @@ static void streamMessageToRow(Row &row,
+        if (oneofDesc) {
+            auto oneofField =
+                    reflection->GetOneofFieldDescriptor(msg, oneofDesc);
 
-                streamMessageToRow(row, reflection->GetMessage(msg, field),
-                                   nestedPrefix);
-            }
-        } else {
-            switch (field->cpp_type()) {
-            case FieldDescriptor::CppType::CPPTYPE_INT32: {
-                auto value = reflection->GetInt32(msg, field);
-                row[name] = value;
-                break;
+            if (oneofField && oneofField == field) {
+                std::string oneofName = prefix + oneofDesc->name();
+                std::string oneofFieldName = prefix + oneofField->name();
+                row[oneofName] = oneofFieldName;
             }
+        }
 
-            case FieldDescriptor::CppType::CPPTYPE_INT64: {
-                auto value = reflection->GetInt64(msg, field);
-                row[name] = value;
-                break;
+        if (field->is_map()) {
+            auto repeated =
+                    reflection->GetRepeatedFieldRef<Message>(msg, field);
+            for (const auto &item : repeated) {
+                streamMapPairToRow(row, item, name);
             }
+        } else if (field->type() == FieldDescriptor::Type::TYPE_MESSAGE) {
+            if (field->is_repeated()) {
+                auto repeated =
+                        reflection->GetRepeatedFieldRef<Message>(msg, field);
+
+                int i = 0;
+                for (const auto &item : repeated) {
+                    std::string nestedPrefix;
+
+                    nestedPrefix = name + "[" + std::to_string(i) + "].";
+                    i++;
+                    streamMessageToRow(row, item, nestedPrefix);
+                }
+            } else {
+                if (!reflection->HasField(msg, field)) {
+                    continue;
+                }
 
-            case FieldDescriptor::CppType::CPPTYPE_UINT32: {
-                auto value = reflection->GetUInt32(msg, field);
-                row[name] = value;
-                break;
+                std::string nestedPrefix = name + ".";
+                streamMessageToRow(row, reflection->GetMessage(msg, field),
+                                   nestedPrefix);
             }
+        } else {
+            streamFieldToRow(row, msg, reflection, field, name);
+        }
+    }
+}
 
-            case FieldDescriptor::CppType::CPPTYPE_UINT64: {
-                auto value = reflection->GetUInt64(msg, field);
-                row[name] = value;
-                break;
-            }
+Row &operator<<(Row &row, const ::google::protobuf::Message &msg) {
+    streamMessageToRow(row, msg, "");
+    return row;
+}
 
-            case FieldDescriptor::CppType::CPPTYPE_DOUBLE: {
-                auto value = reflection->GetDouble(msg, field);
-                row[name] = value;
-                break;
-            }
+static void streamDescriptionToRow(Row &row,
```

Comment:
As we no longer stream Descriptor (using << operator), maybe we should call this setupRowAsHeader or something like that?